src/patternfly/components/FileUpload/file-upload.scss

```diff
@@ -0,0 +1,132 @@
+  --pf-c-file-upload--c-button--m-control--disabled--after--BorderTopColor: var(--pf-global--BorderColor--300);
+  --pf-c-file-upload--c-button--m-control--disabled--after--BorderRightColor: var(--pf-global--BorderColor--300);
+  --pf-c-file-upload--c-button--m-control--disabled--after--BorderBottomColor: var(--pf-global--BorderColor--200);
+  --pf-c-file-upload--c-button--m-control--disabled--after--BorderLeftColor: var(--pf-global--BorderColor--300);
+  --pf-c-file-upload--c-button--m-control--disabled--after--BorderWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-file-upload--c-button--m-control--OutlineOffset: calc(-1 * var(--pf-global--spacer--xs));
+
+  display: flex;
+  flex-direction: column;
+
+  &.pf-m-draganddrop {
+    position: relative;
+    border: var(--pf-c-file-upload--m-draganddrop--BorderWidth) solid var(--pf-c-file-upload--m-draganddrop--BorderColor);
+
+    &::after {
+      position: absolute;
+      top: 0;
+      right: 0;
+      bottom: 0;
+      left: 0;
+      content: "";
+      background-color: var(--pf-c-file-upload--m-draganddrop--after--BackgroundColor);
+      opacity: var(--pf-c-file-upload--m-draganddrop--after--Opacity);
+    }
+  }
+}
+
+.pf-c-input-group > .pf-c-form-control {
+  &::placeholder {
+    color: var(--pf-c-file-upload--c-input-group--c-form-control--placeholder--Color);
+  }
+}
+
+.pf-c-file-upload__text-area {
+  position: relative;
+  min-height: var(--pf-c-file-upload__text-area--MinHeight);
+  padding: var(--pf-c-file-upload__text-area--PaddingTop) var(--pf-c-file-upload__text-area--PaddingRight) var(--pf-c-file-upload__text-area--PaddingBottom) var(--pf-c-file-upload__text-area--PaddingLeft);
+  resize: vertical;
+  background-color: var(--pf-c-file-upload__text-area--BackgroundColor);
+  border: var(--pf-c-file-upload__text-area--BorderWidth) solid;
+  border-color: var(--pf-c-file-upload__text-area--BorderTopColor) var(--pf-c-file-upload__text-area--BorderRightColor) var(--pf-c-file-upload__text-area--BorderBottomColor) var(--pf-c-file-upload__text-area--BorderLeftColor);
+
+  &:not(:read-only) {
+    &.pf-m-hover,
+    &:hover {
+      --pf-c-file-upload__text-area--BorderBottomColor: var(--pf-c-file-upload__text-area--hover--BorderBottomColor);
+    }
+
+    &.pf-m-active,
+    &:active {
+      --pf-c-file-upload__text-area--BorderBottomColor: var(--pf-c-file-upload__text-area--active--BorderBottomColor);
+
+      border-bottom-width: var(--pf-c-file-upload__text-area--active--BorderBottomWidth);
+    }
+
+    &.pf-m-focus,
+    &:focus {
+      --pf-c-file-upload__text-area--BorderBottomColor: var(--pf-c-file-upload__text-area--focus--BorderBottomColor);
+
+      border-bottom-width: var(--pf-c-file-upload__text-area--focus--BorderBottomWidth);
+    }
+  }
+
+  &:read-only,
+  &.pf-m-readonly {
+    --pf-c-file-upload__text-area--BackgroundColor: var(--pf-c-file-upload__text-area--readonly--BackgroundColor);
+  }
+
+  &.pf-m-loading {
```

Comment:
instead of changing this element from a textarea to a div and sharing the same class, if you just make it a div and nest a textarea.pf-c-form-control in it, you can set the textarea's opacity to 0 when it's in the loading state, the absolutely position the spinner in this div as you have in your example.